src/Shared/PackageManagers/MavenProjectManager.cs

```diff
@@ -231,36 +179,45 @@ public override async Task<IEnumerable<string>> EnumerateVersionsAsync(PackageUR
-                string feedUrl = purl?.Qualifiers?["repository_url"] ?? ENV_MAVEN_ENDPOINT.GetRepositoryUrl();
-                MavenSupportedUpstream upstream = feedUrl.GetMavenSupportedUpstream();
+                string feedUrl = purl?.Qualifiers?["repository_url"] ?? ENV_MAVEN_ENDPOINT;
 
                 HttpClient httpClient = CreateHttpClient();
 
-                string baseUrl = string.Empty;
-                if (upstream == MavenSupportedUpstream.MavenCentralRepository)
-                {
-                    baseUrl = $"{upstream.GetRepositoryUrl().EnsureTrailingSlash()}{packageNamespace}/{packageName}/";
-                }
-                else if (upstream == MavenSupportedUpstream.GoogleMavenRepository)
+                var baseUrl = $"{feedUrl.EnsureTrailingSlash()}{packageNamespace}/{packageName}/maven-metadata.xml";
+
+                string? content = string.Empty;
+                try
                 {
-                    baseUrl = $"{upstream.GetDownloadRepositoryUrl().EnsureTrailingSlash()}{packageNamespace}/group-index.xml";
+                    content = await GetHttpStringCache(httpClient, baseUrl, useCache);
                 }
-
-                // Check that the package exists
-                string? content = await GetHttpStringCache(httpClient, baseUrl, useCache);
+                catch {}
 
                 List<string> versionList = new();
-                if (string.IsNullOrWhiteSpace(content))
+                if (!string.IsNullOrWhiteSpace(content))
                 {
-                    return new List<string>();
-                }
+                    // Parse the maven-metadata.xml file for the versions.
+                    XmlDocument xmlDoc = new XmlDocument();
+                    xmlDoc.LoadXml(content);
 
-                // Parse the html file.
-                HtmlParser parser = new();
-                AngleSharp.Html.Dom.IHtmlDocument document = await parser.ParseDocumentAsync(content);
+                    var packageElement = xmlDoc.SelectSingleNode($"//versions");
 
-                if (upstream == MavenSupportedUpstream.MavenCentralRepository)
+                    if (packageElement != null)
+                    {
+                        foreach (XmlNode versionNode in packageElement.ChildNodes)
+                        {
+                            versionList.Add(versionNode.InnerText.Trim());
+                        }
+                    }
+                }
+                else
                 {
+                    // If maven-metadata.xml file is unavailable, try using the web server directory index.
+                    baseUrl = $"{feedUrl.EnsureTrailingSlash()}{packageNamespace}/{packageName}/";
+                    content = await GetHttpStringCache(httpClient, baseUrl, useCache);
+
+                    HtmlParser parser = new();
+                    AngleSharp.Html.Dom.IHtmlDocument document = await parser.ParseDocumentAsync(content);
+
```

Comment:
Could these benefit from the new helper methods?

Reply:
Added the helper methods!